test/e2e/syncer/multishard/multishard_test.go

```diff
@@ -0,0 +1,178 @@
+
+	downstreamWS2NS := syncer.DownstreamNamespaceFor(t, workloadWorkspace2Name, "default")
+	t.Logf("Downstream namespace in workspace 2 is %s", downstreamWS2NS)
+
+	downstreamKubeClient, err := kubernetes.NewForConfig(syncer.DownstreamConfig)
+	require.NoError(t, err)
+
+	framework.Eventually(t, func() (success bool, reason string) {
+		_, err := downstreamKubeClient.AppsV1().Deployments(downstreamWS1NS).Get(ctx, "test1", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, "deployment test1 not found downstream"
+		}
+		require.NoError(t, err)
+		return true, ""
+	}, wait.ForeverTestTimeout, time.Millisecond*500, "deployment test1 not synced downstream")
+
+	framework.Eventually(t, func() (success bool, reason string) {
+		_, err := downstreamKubeClient.AppsV1().Deployments(downstreamWS2NS).Get(ctx, "test2", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, "deployment test2 not found downstream"
+		}
+		require.NoError(t, err)
+		return true, ""
+	}, wait.ForeverTestTimeout, time.Millisecond*500, "deployment test1 not synced downstream")
+
+	if len(framework.TestConfig.PClusterKubeconfig()) == 0 {
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			_, err := downstreamKubeClient.AppsV1().Deployments(downstreamWS1NS).ApplyStatus(ctx, appsv1apply.Deployment("test1", downstreamWS1NS).WithStatus(&appsv1apply.DeploymentStatusApplyConfiguration{
+				Replicas:          ptr(int32(1)),
+				UpdatedReplicas:   ptr(int32(1)),
+				AvailableReplicas: ptr(int32(1)),
+				ReadyReplicas:     ptr(int32(1)),
+			}), metav1.ApplyOptions{FieldManager: "e2e-test", Force: true})
+			return err
+		})
+		require.NoError(t, err)
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			_, err := downstreamKubeClient.AppsV1().Deployments(downstreamWS2NS).ApplyStatus(ctx, appsv1apply.Deployment("test2", downstreamWS2NS).WithStatus(&appsv1apply.DeploymentStatusApplyConfiguration{
+				Replicas:          ptr(int32(1)),
+				UpdatedReplicas:   ptr(int32(1)),
+				AvailableReplicas: ptr(int32(0)),
+				ReadyReplicas:     ptr(int32(0)),
+			}), metav1.ApplyOptions{FieldManager: "e2e-test", Force: true})
+			return err
+		})
+		require.NoError(t, err)
+	}
+
+	framework.Eventually(t, func() (success bool, reason string) {
+		test1Downstream, err := downstreamKubeClient.AppsV1().Deployments(downstreamWS1NS).Get(ctx, "test1", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, "deployment test1 not found downstream"
+		}
+		require.NoError(t, err)
+
+		test1Upstream, err := upstreamKubeClusterClient.AppsV1().Cluster(workloadWorkspace1Path).Deployments("default").Get(ctx, "test1", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, "deployment test1 not found upstream"
+		}
+		require.NoError(t, err)
+		diff := cmp.Diff(test1Downstream.Status, test1Upstream.Status)
+		return len(diff) == 0, fmt.Sprintf("status different between upstream and downstream: %s", diff)
```

Comment:
```suggestion
		return len(diff) == 0, fmt.Sprintf("status different between downstream and upstream: %s", diff)
```

Reply:
Have the text match the order given to cmp.Diff

Reply:
Fixed in commit a4a0c311c02bde305631bc9b28ccdac5a16e3200